### PR TITLE
D 01270 fix signals

### PR DESCRIPTION
**Pull request by JettTech**
### Updates:
 - fixes signals handler
 - adds integration test for signals

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -801,17 +800,17 @@ class Chaperone {
       let event_id = this.createEventId();
       log(`Subscribing to event ${event_id}`);
       this.conn.subscribe(event_id)
-      this.conn.on(event_id, this.sendSignal.bind(this))
+      this.conn.on(event_id, data => {
+        return this.sendSignal(data)
+      })
```
Comment (timotree3) [inline src/index.ts:805]: Why didn't this work before?

Comment (JettTech) [reply src/index.ts:805]: Well, I was wondering why/whether it actually did work in this case before. :joy:  I'm guessing that this code was copied from the hcc case, where the binding is needed, but this case was never tested through to the UI.

Comment (timotree3) [reply src/index.ts:805]: > Well, I was wondering why/whether it actually did work in this case before.
What do you mean? The code before and after this change looks equivalent.`this.sendSignal.bind(this)` should evaluate to a function that calls `this.sendSignal` with whatever is passed to it, exactly the same as the new function you wrote.

**Comment by timotree3** [comment]
I wanted to check if this actually fixed the problem so I checked out this branch, reverted src/index.ts, and ran the tests. They passed. This change seems fine for fixing HCC mode signals but doesn't look like it will affect production hosted happs.

**Review by timotree3 — APPROVED** [review]

**Comment by JettTech** [comment]
Just to recap our offline conversation:
I manually tested this update in a production environment (ie: connected to a holoport and production builds of chaperone and the ui) and it worked with the fix, but not without it.  

However, I just reviewed the unit and integration tests, and indeed, they pass with or without the fix.  I actually see this as more of an issue with the tests than the fix.  In short, I def think it's worth considering writing a more complex test as I don't think it's behavior is a full/true mapping of the production case.  I'll talk to Rob about the prioritization of this.

**Pull request merged**